controllers/clustergroupupgrade_controller.go

```diff
@@ -573,6 +594,9 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 				utils.TimeoutMessages[clusterGroupUpgrade.RolloutType()],
 			)
 			err = r.handleBatchTimeout(ctx, clusterGroupUpgrade)
+
+			r.sendEventCGUBatchUpgradeTimedout(clusterGroupUpgrade)
```

Comment:
Why not make it part of handleBatchTimeout()?

Reply:
I'll explore the conditional sending of this event inside handleBatchTimeout.